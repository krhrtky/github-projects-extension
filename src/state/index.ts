import { createSignal } from "solid-js";

export const menuOpened = createSignal(false);
