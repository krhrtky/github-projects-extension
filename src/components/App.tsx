import { menuOpened } from "../state";
import { Menu } from "./Menu";
import { Content } from "./Content";
import style from "./App.module.scss";

export const App = () => {
  const [value, ] = menuOpened;

  return (
    <div class={style.app}>
      { value() ? <Content /> : <Menu /> }
    </div>
  );
}
