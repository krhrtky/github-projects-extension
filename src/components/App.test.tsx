import { describe, it, expect, afterEach } from "vitest";
import { render, cleanup } from "solid-testing-library";
import { App } from "./App";
import { menuOpened } from "../state";

describe(App.name, () => {
  afterEach(cleanup);

  it("is menu opened", () => {
    const [, setState] = menuOpened;
    setState(() => true);

    const { container } = render(() => <App/>);
    expect(container.innerHTML).toMatchSnapshot();
  });

  it("is not menu opened", () => {
    const [, setState] = menuOpened;
    setState(() => false);

    const { container } = render(() => <App/>);
    expect(container.innerHTML).toMatchSnapshot();
  });
});
