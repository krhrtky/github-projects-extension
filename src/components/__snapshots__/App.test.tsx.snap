// Vitest Snapshot v1

exports[`_Hot$$App > is menu opened 1`] = `"<div style=\\"_app_f68c22\\"><div class=\\"Box d-flex flex-column anim-fade-in fast\\" style=\\"width: 200px;\\"><div class=\\"Box-header\\"><button class=\\"Box-btn-octicon btn-octicon float-right\\" type=\\"button\\" aria-label=\\"Close dialog\\"><svg version=\\"1.1\\" width=\\"16\\" height=\\"16\\" viewBox=\\"0 0 16 16\\" class=\\"octicon octicon-x\\" aria-hidden=\\"true\\"><path fill-rule=\\"evenodd\\" d=\\"M3.72 3.72a.75.75 0 011.06 0L8 6.94l3.22-3.22a.75.75 0 111.06 1.06L9.06 8l3.22 3.22a.75.75 0 11-1.06 1.06L8 9.06l-3.22 3.22a.75.75 0 01-1.06-1.06L6.94 8 3.72 4.78a.75.75 0 010-1.06z\\"></path></svg></button><h3 class=\\"Box-title\\">Columns</h3></div><div class=\\"overflow-auto\\"><ul></ul></div></div></div>"`;

exports[`_Hot$$App > is not menu opened 1`] = `"<div style=\\"_app_f68c22\\"><button class=\\"btn\\" type=\\"button\\"><svg version=\\"1.1\\" width=\\"16\\" height=\\"16\\" viewBox=\\"0 0 16 16\\" class=\\"octicon octicon-three-bars\\" aria-hidden=\\"true\\"><path fill-rule=\\"evenodd\\" d=\\"M1 2.75A.75.75 0 011.75 2h12.5a.75.75 0 110 1.5H1.75A.75.75 0 011 2.75zm0 5A.75.75 0 011.75 7h12.5a.75.75 0 110 1.5H1.75A.75.75 0 011 7.75zM1.75 12a.75.75 0 100 1.5h12.5a.75.75 0 100-1.5H1.75z\\"></path></svg></button></div>"`;
